### PR TITLE
More mingw patches

### PR DESCRIPTION
**Pull request by inactive123**
Part 2 of mingw patches.

### CODE REVIEW

File: Common/CommonTypes.h [hunk 1]
```diff
@@ -30,7 +30,7 @@ struct u24_be {
 	}
 };
 
-#ifdef _WIN32
+#if defined(_WIN32) && !defined(__MINGW32__)
```
Comment (unknownbrackets) [inline Common/CommonTypes.h]: I think _MSC_VER is more correct here.

-[Unknown]

File: Core/HLE/proAdhoc.h [hunk 2]
```diff
@@ -50,7 +50,7 @@ class PointerWrap;
 #include <fcntl.h>
 #include <errno.h>
 #endif
-#ifdef _MSC_VER
+#ifdef _WIN32
 #define PACK
```
Comment (unknownbrackets) [inline Core/HLE/proAdhoc.h]: Hmm, I would think PACK would be _MSC_VER related.  The others look win32 related.

I think `#define PACK __attribute__((packed))` would be needed when _WIN32 but not _MSC_VER right?

-[Unknown]

File: ext/native/file/file_util.cpp [hunk 3]
```diff
@@ -85,13 +83,23 @@ uint64_t GetSize(FILE *f)
...
+#ifdef _WIN32
+	uint64_t pos = _ftelli64(f);
+#else
 	uint64_t pos = ftello(f);
+#endif
 	if (fseek(f, 0, SEEK_END) != 0) {
 		return 0;
 	}
+#ifdef _WIN32
+	uint64_t size = _ftelli64(f);
```
Comment (unknownbrackets) [inline ext/native/file/file_util.cpp]: Hmm, what are we gaining here by duplicating the code and making it so there's an extra place to miss when fixing bugs?

Not saying this is bad, just "more mingw patches" isn't very descriptive.  6 months from now I may see this code and "fix it", because DRY.

-[Unknown]

**Review by unknownbrackets — CHANGES_REQUESTED** [review]
A couple notes, but LGTM.

-[Unknown]

File: Common/FileUtil.cpp [hunk 4]
```diff
@@ -93,7 +93,7 @@ FILE *OpenCFile(const std::string &filename, const char *mode)
 
 bool OpenCPPFile(std::fstream & stream, const std::string &filename, std::ios::openmode mode)
 {
-#if defined(_WIN32) && defined(UNICODE)
+#if defined(_WIN32) && defined(UNICODE) && !defined(__MINGW32__)
```
Comment (hrydgard) [inline Common/FileUtil.cpp]: Hm, why is this needed? No wstring support in MingW32's libraries?

(ditto for the other places)

Comment (twinaphex) [reply Common/FileUtil.cpp]: It's odd yeah - I am even compiling with UNICODE defines.

File: ext/native/file/file_util.cpp (hunk 3 above)
Comment (twinaphex) [reply ext/native/file/file_util.cpp]: It's necessary to work around an error. The macro will clash with the mingw headers that provide prototypes for _ftelli64/_fseeki64, etc.

File: Common/CommonTypes.h (hunk 1 above)
Comment (twinaphex) [reply Common/CommonTypes.h]: True, I will change that.

File: ext/native/file/file_util.cpp (hunk 3 above)
Comment (hrydgard) [reply ext/native/file/file_util.cpp]: Well, at least comment it appropriately.

File: Common/FileUtil.cpp (hunk 4 above)
Comment (twinaphex) [reply Common/FileUtil.cpp]: This is the error I get when I take out && !defined(__MINGW32__) here -

https://hastebin.com/bucebodoci.vbs

**Comment by hrydgard** [comment]
Something went wrong with the PACK define on non-Windows: 

https://travis-ci.org/hrydgard/ppsspp/builds/270148718?utm_source=github_status&utm_medium=notification

**Comment by twinaphex** [comment]
I'll change it back to what it was before, and I'll leave it up to you to delete it later on if you choose to do that.

**Comment by hrydgard** [comment]
Yeah, let's deal with PACK separately.
